apps/system/js/app_window.js

```diff
@@ -224,14 +224,14 @@
     this.getTopMostWindow()._setActive(enable);
   };
 
-  AppWindow.prototype.inScope = function(scope) {
+  AppWindow.prototype.inScope = function(scope, name) {
```

Comment:
"In scope" is a formally defined concept in the W3C Web App Manifest specification http://w3c.github.io/manifest/#navigation-scope with an algorithm based entirely on URLs.

This inScope() method taking "name" as an argument effectively makes scope a tuple of URL scope + name. If a URL scope is provided by a web manifest then being in scope shouldn't be dependent on any app name. There could be a combination of a manifest's scope and an application-name meta tag which break this in a non-spec-compliant way.

If we need a function to check for origin + name as a fallback for scope then let's make that a separate function and call it something else. Rather than muddying the concept of "inScope" on an app window I would prefer us to not set the origin as a default value for the scope property (as the spec doesn't say to do this), but instead store a separate .origin property on the siteObject alongside the .name property. We could then have a separate method with a name like "matchesOriginAndName" for the use case of re-using open windows.

How does that sound?